apps/studio/src/store/modules/TabModule.ts

```diff
@@ -50,9 +52,17 @@ export const TabModule: Module<State, RootState> = {
     },
     remove(state, tab: OpenTab) {
       state.tabs = _.without(state.tabs, tab)
+      state.lastClosedTabs.push(tab)
     },
     setActive(state, tab?: OpenTab) {
       state.active = tab
+    },
+    async reopenLastClosedTab(state){
+      const lastClosedTab = state.lastClosedTabs.pop()
+      if(lastClosedTab){
+        state.tabs.push(lastClosedTab)
```

Comment:
Hey! Unfortunately once the tab has been removed it has lost it's ID, and has been deleted from the database -- see actions -> remove.

Probably you'll need to make a copy of it and send it through actions/add to make sure it's saved to the database